test/cases/associations/autosave_association_bug_reproduction_test.rb

```diff
@@ -0,0 +1,41 @@
+require 'cases/helper'
+
+module ActiveRecord
+  module Precount
+    module AutosaveAssociationExtension
+      private
+
+      def save_belongs_to_association(reflection)
+        super(reflection)
+      end
+    end
+  end
+  Base.prepend(Precount::AutosaveAssociationExtension)
+end
+
+class AutosaveAssociationBugReproductionTest < ActiveRecord::CountLoader::TestCase
+  def setup
+    tweets_count.times do |i|
+      tweet = Tweet.create
+      i.times do |j|
+        Favorite.create(tweet: tweet, user_id: j + 1)
+      end
+    end
+  end
+
+  def teardown
+    [Tweet, Favorite].each(&:delete_all)
+  end
+
+  def tweets_count
+    3
+  end
+
+  def test_reproduce_bug_with_save_belongs_to_association_method
+    tweet = Tweet.first
+    tweet.my_favorites_count
+    assert_raises(NoMethodError) do
```

Comment:
Why do you test that this plugin is broken? This line should call `assert_nothing_raised`, shouldn't it?

Since this test passes, current patch does not work. At least I confirmed that our `save_belongs_to_association` is not called (by setting binding.pry inside it).